play/plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherPlugin.kt

```diff
@@ -173,7 +173,7 @@ internal class PlayPublisherPlugin : Plugin<Project> {
         (android as ExtensionAware).extensions.add(PLAY_CONFIGS_PATH, extensionContainer)
 
         val androidExtension = project.extensions.getByType<ApplicationAndroidComponentsExtension>()
-        androidExtension.onVariants v@{ variant ->
+        androidExtension.onVariants(androidExtension.selector().all()) v@{ variant ->
```

Comment:
This is an undocumented breaking change of AGP 8 that I faced before. Because Kotlin defaultImpls works, if the plugin is compiled using AGP `compileOnly` dependency, the byte generated lack of some impl method causing `NoSuchMethodError`, when the consumer AGP version is lower than 8.

I'm not sure if there is any flag in the compiler plugin to avoid this issue, any suggestion is welcome :)